homeassistant/components/husqvarna_automower/coordinator.py

```diff
@@ -26,14 +28,32 @@ def __init__(
             hass,
             _LOGGER,
             name=DOMAIN,
+            update_interval=timedelta(seconds=300),
         )
-        self.session = aioautomower.AutomowerSession(
+        self.mowersession = aioautomower.AutomowerSession(
             implementation.client_id,
             session.token,
             low_energy=False,
             handle_token=False,
+            handle_rest=False,
         )
 
     async def _async_update_data(self) -> None:
-        """Fetch data from Husqvarna."""
-        await self.session.connect()
+        """Subscribe for websocket and poll data from the API."""
+        await self.mowersession.connect()
+        self.mowersession.register_data_callback(
+            self.callback, schedule_immediately=True
+        )
+        return await self.mowersession.get_status()
+
+    @callback
+    def callback(self, ws_data: aioautomower.MowerData):
+        """Process websocket callbacks and write them to the DataUpdateCoordinator."""
+        if self.data:
+            for mower in self.data.data:
+                if mower.id == ws_data.id:
+                    for field in fields(ws_data.attributes):
```

Comment:
I suggest making a helper method on the `mower` class in the library that takes `ws_data` and updates the mower attributes.

Reply:
But then I would have to pass the polled data from the coordinator to the library. Wouldn't that be too complicated, without a real benefit?

Reply:
No, I suggest we pass the websocket update data, ie the `MowerData` instance  to a method on the existing stored `MowerData` instance. The benefit is that the library takes care of the details of how to apply the updated websocket data.

Alternatively since the library already stores the `MowerData` instances on the client, the websocket could update them directly and just let the library user know which mower was updated.

Also, I suggest storing the `MowerData` instances in a dict instead of a list, if the `id` of the mower data instance is unique. That way we can easily look up a specific mower data instance if we have the id.

Reply:
Okay, got it.